src/Workspaces/CSharp/Portable/Utilities/CSharpTypeStyleHelper.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.CSharp.Utilities
+{
+    internal struct TypeStyleResult
+    {
+        private readonly CSharpTypeStyleHelper _helper;
+        private readonly TypeSyntax _typeName;
+        private readonly SemanticModel _semanticModel;
+        private readonly OptionSet _optionSet;
+        private readonly CancellationToken _cancellationToken;
+
+        /// <summary>
+        /// Whether or not converting would transition the code to the style the user prefers. i.e.
+        /// if the user likes 'var' for everything, and you have 'int i = 0' then IsStylePreffered
+        /// will be true.  however, if the user likes 'var' for everything and you have 'var i = 0',
+        /// then it's still possible to convert that, it would just be 'false' for IsStylePreferred
+        /// because it goes against the user's preferences.
+        ///
+        /// In general, most features should only convert the type if IsStylePreferred is true.  The
+        /// one exception is the refactoring, which is explicitly there to still let people convert
+        /// things quickly, even if it's going against their stated style.
+        /// </summary>
+        public readonly bool IsStylePreferred;
```

Comment:
The doc comment is very helpful, but the naming `IsStylePreferred` still seems a bit confusing. I would have assumed that `AnalyzeTypeName(typeSyntax).IsStylePreferred` would tell me if the _current_ type style of `typeSyntax` is preferred, but the current implementation of the flag is opposite. Should this be renamed to `IsConvertedStylePreferred` with current semantics or to `IsExistingTypeStylePreferred` with negated semantics?

Reply:
This code moved down. I didn't want to change it.  :)   I can change the name if you really want though.